client/state/selectors/get-simple-payments.js

```diff
@@ -25,7 +26,11 @@ export default createSelector(
 		const simplePaymentProducts = get( state, `simplePayments.productList.items.${ siteId }`, null );
 
 		if ( ! simplePaymentId ) {
-			return simplePaymentProducts;
+			if ( ! simplePaymentProducts ) {
+				return null;
+			}
```

Comment:
nit: this condition could be moved outside the `if ( ! simplePaymentId )` -- right after the `get`.